pkg/controller/tektoninstallation/tektoninstallation_controller_test.go

```diff
@@ -0,0 +1,171 @@
+package tektoninstallation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/codeready-toolchain/toolchain-operator/pkg/apis"
+	"github.com/codeready-toolchain/toolchain-operator/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-operator/pkg/tekton"
+	"github.com/codeready-toolchain/toolchain-operator/pkg/test"
+	. "github.com/codeready-toolchain/toolchain-operator/pkg/test/olm"
+	. "github.com/codeready-toolchain/toolchain-operator/pkg/test/toolchain"
+	olmv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"testing"
+)
+
+func TestTektonInstallationController(t *testing.T) {
+	t.Run("should reconcile with tekton installation", func(t *testing.T) {
+		// given
+		tektonSub := tekton.NewSubscription(tekton.SubscriptionNamespace)
+		tektonInstallation := NewTektonInstallation()
+		cl, r := configureClient(t, tektonInstallation)
+		request := newReconcileRequest(tektonInstallation)
+
+		t.Run("should create tekton subscription and requeue", func(t *testing.T) {
+			// when
+			_, err := r.Reconcile(request)
+
+			// then
+			require.NoError(t, err)
+
+			AssertThatTektonInstallation(t, tektonInstallation.Namespace, tektonInstallation.Name, cl).
+				HasConditions(tekton.SubscriptionCreated(tekton.SubscriptionSuccess))
```

Comment:
We can add `AssertThatSubscription()` here too ??

Reply:
Addressed in  40cbf9c